File: c/questao4.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

int compararStringLista(const char *string, const char *lista[], int tamanho) {
    for (int i = 0; i < tamanho; i++) {
        if (strcmp(string, lista[i]) == 0) {
            return 1;
        }
    }
    return 0;
}

int main() {
    int numNomes;

    printf("Digite o número de nomes na lista: ");
    scanf("%d", &numNomes);

    const char *nomes[numNomes];

    for (int i = 0; i < numNomes; i++) {
        nomes[i] = malloc(50);
        printf("Digite o nome #%d: ", i + 1);
        scanf("%s", (char *)nomes[i]);
    }

    char stringBuscada[50];

    printf("Digite uma string para buscar na lista de nomes: ");
    scanf("%s", stringBuscada);

    if (compararStringLista(stringBuscada, nomes, numNomes)) {
        printf("String encontrada na lista.\n");
    } else {
        printf("String não encontrada na lista.\n");
    }

    for (int i = 0; i < numNomes; i++) {
        free((char *)nomes[i]);
    }

    return 0;
}

File: python/questao4.py
def comparar_string_lista(string, lista):
    return string in lista

# Solicita ao usuário a lista de nomes
num_nomes = int(input("Digite o número de nomes na lista: "))
nomes = []

for i in range(num_nomes):
    nome = input(f"Digite o nome #{i + 1}: ")
    nomes.append(nome)

string_buscada = input("Digite uma string para buscar na lista de nomes: ")

if comparar_string_lista(string_buscada, nomes):
    print("String encontrada na lista.")
else:
    print("String não encontrada na lista.")